proversity_reports_script/report_backend/video_completion_report.py

```diff
@@ -0,0 +1,208 @@
+
+        if not user_list_file_name:
+            print('User list file name was not provided.')
+            exit()
+
+        local_path_name = '{parent_folder}/result/{file_name}.csv'.format(
+            parent_folder=os.path.join(os.path.dirname(__file__), os.pardir),
+            file_name=now.strftime('%Y-%m-%d'),
+        )
+
+        print('Downloading user list report {} into {}'.format(user_list_file_name, local_path_name))
+
+        report_bucket.download_file(user_list_file_name, local_path_name)
+
+        return local_path_name
+
+
+def generate_csv_dict(course_data={}, user_list_data=[]):  # pylint: disable=dangerous-default-value
+    """
+    Return a csv dict to write the csv file.
+    """
+    csv_data = []
+
+    for user_data in course_data:
+        email = user_data.get('email', '')
+
+        if not email in user_list_data:
+            continue
+
+        first_name = user_data.get('first_name', '')
+        last_name = user_data.get('last_name', '')
+        course_is_complete = user_data.get('course_is_complete', '')
+        student_enrollment_id = user_data.get('student_enrollment_id', '')
+
+        dict_writer_data = {}
+
+        dict_writer_data['First Name'] = first_name
+        dict_writer_data['Last Name'] = last_name
+        dict_writer_data['Student Enrollment ID'] = student_enrollment_id
+        dict_writer_data['Email'] = email
+        dict_writer_data['Course Is Complete'] = course_is_complete
+
+        dict_writer_data.update(get_required_activity_dict(user_data))
+
+        csv_data.append(dict_writer_data)
+
+    return csv_data
+
+
+def get_required_activity_dict(user_data):
+    """
+    Create a dict with the required activity data.
+
+    Args:
+        user_data: Report json data per course.
+    Returns:
+        Dict containing activities info.
+        {
+            'Multiple Choice': 'completed',
+            'Image Explorer': 'completed',
+            'Video': 'not_completed'
+        }
+    """
+    required_activities_data = {}
+    total_activities = user_data.get('total_activities', 0)
+
+    if total_activities:
+        total = int(total_activities)
+        # Create as many 'required_activity_' as the total number of activities.
+        for activity_number in range(1, total + 1): # Plus 1, because the stop argument it's not inclusive.
+            required_activity_number = user_data.get('required_activity_{}'.format(activity_number), '')
+            required_activity_name = user_data.get('required_activity_{}_name'.format(activity_number), '')
+
+            # Let's add the activity number at the end of the name if two or more activities have the same name.
+            if required_activity_name in required_activities_data.keys():
+                required_activity_name = '{}-{}'.format(
+                    required_activity_name,
+                    activity_number,
+                )
+
+            required_activities_data.update({
```

Comment:
Same comment as above

Reply:
Let's keep this as it.

Reply:
why ? if probably this will be modified in the future.

Reply:
Yes, maybe we need to add more fields.

Reply:
my mistake if probably this won't be modified in the future.

Reply:
This it's not about the line too long, it's about the code reading.